gamemodes/terrortown/gamemode/client/cl_voice.lua

```diff
@@ -201,112 +182,70 @@ function GM:PlayerStartVoice(ply)
         return
     end
 
-    local client = LocalPlayer()
-
-    VOICE.UpdatePlayerVoiceVolume(ply)
-
-    if not IsValid(g_VoicePanelList) or not IsValid(client) then
-        return
-    end
-
-    -- There'd be an extra one if voice_loopback is on, so remove it.
-    GAMEMODE:PlayerEndVoice(ply, true)
+    local plyRoleData = ply:GetSubRoleData()
 
-    -- Tell server this is global
-    if client == ply then
-        local tm = client:GetTeam()
+    local client = LocalPlayer()
+    local clientTeam = client:GetTeam()
+    local clientRoleData = client:GetSubRoleData()
 
-        local isGlobal = not VOICE.isTeam
-        client[tm .. "_gvoice"] = isGlobal
+    if ply ~= client then
+        -- if someone else than us starts talking, the UI should be updated as well
+        ply:SetSpeakingInVoice(true)
+    else
+        -- todo: is this used anywhere? why?
+        client[client:GetTeam() .. "_gvoice"] = not VOICE.isTeam
 
+        -- notify server this if this is a global voice chat
         net.Start("TTT2RoleGlobalVoice")
-        net.WriteBool(isGlobal)
+        net.WriteBool(not VOICE.isTeam)
         net.SendToServer()
-
-        VOICE.SetSpeaking(true)
-    end
-
-    local pnl = g_VoicePanelList:Add("VoiceNotify")
-    pnl:Setup(ply)
-    pnl:Dock(TOP)
-
-    local oldThink = pnl.Think
-
-    pnl.Think = function(s)
-        oldThink(s)
-
-        VoiceNotifyThink(s)
     end
 
-    local shade = Color(0, 0, 0, 150)
-
-    -- TODO recreate all voice panels on HUD switch
-    local paintFn = function(s, w, h)
-        if not IsValid(s.ply) then
-            return
-        end
-
-        draw.RoundedBox(4, 0, 0, w, h, s.Color)
-        draw.RoundedBox(4, 1, 1, w - 2, h - 2, shade)
-    end
-
-    if huds and HUDManager then
-        local hud = huds.GetStored(HUDManager.GetHUD())
-        if hud then
-            paintFn = hud.VoicePaint or paintFn
```

Comment:
Can you remove this from the oldttt hud if it is defined?

Reply:
it wasn't defined for oldTTT as this was only a hacky overwrite